src/gui/labeling/qgslabelinggui.cpp

```diff
@@ -962,8 +996,9 @@ void QgsLabelingGui::validateGeometryGeneratorExpression()
 
 void QgsLabelingGui::determineGeometryGeneratorType()
 {
-  if ( !mPreviewFeature.isValid() && mLayer )
-    mLayer->getFeatures( QgsFeatureRequest().setLimit( 1 ) ).nextFeature( mPreviewFeature );
+  QgsVectorLayer *vLayer = qobject_cast< QgsVectorLayer * >( mLayer );
+  if ( !mPreviewFeature.isValid() && vLayer )
```

Comment:
```suggestion
  QgsVectorLayer *vLayer = qobject_cast< QgsVectorLayer * >( mLayer );
  if ( !vLayer )
    return;
    
  if ( !mPreviewFeature.isValid() && vLayer )
```

Reply:
Same here, `vLayer` is checked for before being accessed anyway